patching libsparse.mk to build in yocto environment.

Changes include:
Introduced an output folder which will keep all the outputs from the
android-tools.

Adding an include file that is setting the common flags.
Changed the cppflag to include the headers from other components
of android-tools

Modified LDflags so that the libraries that are coming from other android-tools
are searched in the output folder.

Modified the build rule so that it outputs the binary to the OUT_DIR

Signed-off-by: Nisha Parrakat <Nisha.Parrakat@kpit.com>

Upstream-Status: Pending

--- git/system/core/debian/libsparse.mk	2021-05-17 08:13:34.764584388 +0000
+++ git/system/core/debian/libsparse.mk	2021-05-17 08:15:34.850492654 +0000
@@ -1,3 +1,4 @@
+include ../../rules_yocto.mk
 NAME = libsparse
 SOURCES = \
         backed_block.cpp \
@@ -18,16 +19,16 @@
   -shared -Wl,-soname,$(NAME).so.0 \
   -lz \
   -Wl,-rpath=/usr/lib/$(DEB_HOST_MULTIARCH)/android \
-  -L. \
+  -L$(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/ \
   -lbase
 
 
 build: $(COBJECTS) $(CXXOBJECTS)
-	$(CXX) $^ -o $(NAME).so.0 $(LDFLAGS)
-	ln -s $(NAME).so.0 $(NAME).so
+	$(CXX) $^ -o $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so.0 $(LDFLAGS)
+	ln -s $(NAME).so.0 $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so	
 
 clean:
-	$(RM) $(CXXOBJECTS) $(COBJECTS) $(NAME).so*
+	$(RM) $(CXXOBJECTS) $(COBJECTS) $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so*
 
 $(COBJECTS): %.o: %.c
 	$(CC) $< -o $@ $(CFLAGS) $(CPPFLAGS)
